### PR TITLE
1. DataAccessTokens download token details on auth. portals

### PR DESCRIPTION
**Pull request by ao508**
~Closed PR #1826 (dat feature branch to rc) and opened this PR to `master` instead.~
~Closed PR #1839 (dat feature branch to master) and opened this PR to `release-3.0.0` instead.~
~Closed PR #2264 (dat feature branch to release-3.0.0) and opened this PR to `release-3.1.0` instead)
cc @alisman, @sheridancbio 

# What? Why?

1. DataAccessTokens download token details on authenticated portals
* Added DAT endpoint calls to `CBioPortalAPI.ts`
* Added new react component for DataAccessTokens
* Moved sign out link to dropdown
* Add DAT file contents build and download link
* Suppress DAT download if auth method is social_auth

2. Migrate DAT endpoint to main API

3. Add DAT method check to hide 'Download Token' option if DAT method is 'none'
* Added check to determine whether DAT method is "none"
* If DAT method is "none" then hide 'Token Download' from authorized user dropdown menu.

4. Format DAT dates for download token file (ISO format)

## Example login dropdown menu:
dat-dropdown-authenticated
Signed-off-by: Angelica Ochoa <aochoa4230@gmail.com>

## Example token file: 
[cbioportal_data_access_token.txt](https://github.com/cBioPortal/cbioportal-frontend/files/2681707/cbioportal_data_access_token.txt)

# Checks
- [ ] Is this PR adding logic based on one or more **clinical** attributes? If yes, please make sure validation for this attribute is also present in the data validation / data loading layers (in backend repo) and documented in [File-Formats Clinical data section](https://github.com/cBioPortal/cbioportal/blob/master/docs/File-Formats.md#clinical-data)!
- [ ] Follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [ ] Follows the [Airbnb React/JSX Style guide](https://github.com/airbnb/javascript/tree/master/react).
- [ ] Make sure your commit messages end with a Signed-off-by string (this line
  can be automatically added by git if you run the `git-commit` command with
  the `-s` option)

# Notify reviewers
Read our [Pull request merging
policy](../CONTRIBUTING.md#pull-request-merging-policy). If you are part of the
cBioPortal organization, notify the approprate team (remove inappropriate):

@cBioPortal/frontend

If you are not part of the cBioPortal organization look at who worked on the
file before you. Please use `git blame <filename>` to determine that
and notify them here:

### CODE REVIEW

File: src/shared/components/dataAccessTokens/DataAccessTokensDropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+                action:<a href={buildCBioPortalPageUrl(this.props.appStore.logoutUrl)}>Sign out</a>,
+                hide:false
+            },
+            {
+                id:"datDownload",
+                action:<a onClick={() => this.downloadDataAccessTokenFile()}>Download token</a>,
+                hide:(AppConfig.serverConfig.authenticationMethod === "social_auth" || (AppConfig.serverConfig.dat_method !== "uuid" && AppConfig.serverConfig.dat_method !== "jwt"))
+            }
+        ];
+        const shownListItems = listItems.filter((l)=>{
```
Comment (alisman) [inline src/shared/components/dataAccessTokens/DataAccessTokensDropdown.tsx]: this seems kinda weird.  it's hardcoded to false, so wouldn't it just never show?

Comment (ao508) [reply src/shared/components/dataAccessTokens/DataAccessTokensDropdown.tsx]: @alisman it's getting the inverse. If hide is `false` then `return !false` would be `true`

Comment (alisman) [reply src/shared/components/dataAccessTokens/DataAccessTokensDropdown.tsx]: ah. right you are.  you could accomplish this with the new react fragment markup (<></> which allows you to return a list of things without a parent.  you could then just do some plain conditional logic to accomplish this a little more simply.  not important though if you want to leave it like this.

File: src/shared/components/dataAccessTokens/DataAccessTokensDropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        const dat = await this.generateNewDataAccessToken();
+        if (!isNullOrUndefined(dat)) {
+            const fileContents = buildDataAccessTokenFileContents(dat);
+            fileDownload(fileContents, "cbioportal_data_access_token.txt");
+        }
+    }
+
+    async generateNewDataAccessToken() {
+        if (this.props.appStore.isLoggedIn) {
+            let _token = await Promise.resolve(
```
Comment (alisman) [inline src/shared/components/dataAccessTokens/DataAccessTokensDropdown.tsx]: I don't think you need Promise.resolve here.  the client method returns a promise which you can await directly.

**Comment by alisman** [comment]
Looks like there's a build [failure](https://circleci.com/gh/cBioPortal/cbioportal-frontend/42271?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link).  I think it has to do with generation of the internal client.   You could try running the updateAPI with appropriate backend specified.  If that doesn't fix issue then it's possible that the swagger doc is broken, which would indicate some issue with api itself, since I think swagger doc is itself generated from actual api

**Comment by ao508** [comment]
@alisman update and rebased the pr again

**Comment by alisman** [comment]
@ao508 there's still a build issue `[going on](https://dashboard.heroku.com/pipelines/4e51d3c7-887c-4538-8d7e-e6c26cfd0814/review-app/acdb2f4b-3627-44bc-9207-d67cdc0ea080)`.  Do you see this locally?  Let me know if you need some help resolving it.

**Comment by ao508** [comment]
@alisman looks good! thanks for the help! 
Screen Shot 2019-07-16 at 4 18 22 PM

**Comment by ao508** [comment]
@alisman everything except a screen shot test failed. Think it's fine to merge?

**Pull request merged**